apps/server/src/modules/room/api/room.uc.ts

```diff
@@ -1,51 +1,68 @@
 import { ConfigService } from '@nestjs/config';
-import { Injectable } from '@nestjs/common';
+import { ForbiddenException, Injectable } from '@nestjs/common';
 import { Page } from '@shared/domain/domainobject';
 import { EntityId } from '@shared/domain/types';
-import { IFindOptions } from '@shared/domain/interface';
+import { IFindOptions, Permission, RoleName } from '@shared/domain/interface';
 import { FeatureDisabledLoggableException } from '@shared/common/loggable-exception';
+import { RoomMemberService } from '@src/modules/room-member';
+import { Action, AuthorizationService } from '@src/modules/authorization';
 import { Room, RoomCreateProps, RoomService, RoomUpdateProps } from '../domain';
 import { RoomConfig } from '../room.config';
 
 @Injectable()
 export class RoomUc {
 	constructor(
 		private readonly configService: ConfigService<RoomConfig, true>,
-		private readonly roomService: RoomService
+		private readonly roomService: RoomService,
+		private readonly roomMemberService: RoomMemberService,
+		private readonly authorizationService: AuthorizationService
 	) {}
 
 	public async getRooms(userId: EntityId, findOptions: IFindOptions<Room>): Promise<Page<Room>> {
 		this.checkFeatureEnabled();
-
-		// TODO check authorization
-		// const user: User = await this.authorizationService.getUserWithPermissions(userId);
-
 		const rooms = await this.roomService.getRooms(findOptions);
-		return rooms;
+		const user = await this.authorizationService.getUserWithPermissions(userId);
+		const authorizationList = await this.roomMemberService.batchHasAuthorization(
+			rooms.data.map((room) => room.id),
+			user,
+			Action.read
+		);
+		const permittedRoomSet = new Set<EntityId>();
+		authorizationList.forEach(({ hasAuthorization, roomId }) => {
+			if (hasAuthorization) permittedRoomSet.add(roomId);
+		});
+		const authorizedRooms = rooms.data.filter((room) => permittedRoomSet.has(room.id));
+
+		// TODO: must find a way to pagainate correctly over the authorized roomIds
+		return new Page(authorizedRooms, authorizedRooms.length);
 	}
 
 	public async createRoom(userId: EntityId, props: RoomCreateProps): Promise<Room> {
 		this.checkFeatureEnabled();
 
-		// TODO check authorization
-
+		const user = await this.authorizationService.getUserWithPermissions(userId);
+		// NOTE: currently only teacher are allowed to create rooms. Could not find simpler way to check this.
+		this.authorizationService.checkOneOfPermissions(user, [Permission.TASK_DASHBOARD_TEACHER_VIEW_V3]);
```

Comment:
surely you can use a permission that has something to do with the operation we are doing?
Either introduce new permissions like ROOM_CREATE, or at the very least use something connected like COURSE_CREATE, assuming that everyone that may create a course is also allowed to create a room. though a new permission is most definetly cleaner.